### ISSUE TITLE
What will it take to get this production ready?

### ISSUE THREAD

**Issue by ClimbsRocks** [dask/dask-searchcv#8]
This is exactly what I've been looking for (ok, looking for is too strong- half-assedly-hacking-together-pieces-of-this-myself is probably a more accurate description). 

I'm running into a bunch of the issues you call out with sklearn's native pipelines. 

I'd love to integrate this into [auto_ml](https://github.com/ClimbsRocks), which we use in production at DoorDash. So I'm wondering what it would take for you to happily certify this production ready?

**Comment by mrocklin** [comment]
Hi @ClimbsRocks .  Most of the work here is done by Dask, the underlying execution system, which is currently in production in several places.  

The algorithms within the dask-learn subproject are at varying levels of maturity.  @jcrist would know more, but I suspect that if all you're interested in is Pipeline and GridSearch then you'll probably be mostly happy with what's here.

My main hesitation would be that this project doesn't have an active user community, so you would be on fresh ground.  Historically, this project comes out of collaboration between Dask and SKLearn developers to demonstrate what the two projects could do together.  Because it didn't have a real-world application behind it there are likely to be a couple of usability rough edges.  My guess is that you would cut yourself a couple of times but that we could easily smooth over those issues.

The entire GridSearch + Pipeline codebase is only 250 LOC on top of the Dask scheduler, so development and maintenance here is cheap.

Your auto-ml looks very cool.  You may also be interested in the generic parallelism techniques available through [dask.delayed](http://dask.pydata.org/en/latest/delayed-overview.html).

**Comment by jcrist** [comment]
Sorry for the late response here. As Matt stated above, this was mostly an experiment in combining dask & scikit-learn. I wouldn't call anything in here "production ready", it hasn't seen enough use to iron out all the inevitable bugs that software entails. That said, the model-parallel Pipeline and GridSearch stuff is pretty solid and complete. You may also be interested in using the [distributed backend to joblib](http://distributed.readthedocs.io/en/latest/joblib.html).

I'll also echo Matt's suggestion to use [dask.delayed](http://dask.pydata.org/en/latest/delayed-overview.html) directly - my experience has been that it's fairly straightforward to add some amount of parallelism to an existing codebase by adding delayed in a few (well thought out) places.

**Comment by RPrudden** [comment]
Hi all. I'm currently working on applying distributed machine learning to meteorology data. We're already using Dask for some analysis, so this library could be particularly useful. In short, I'd be interested in acting as a test user for the project.

**Comment by jcrist** [comment]
This library has been released under `dask-searchcv`, and is now available under pypi and (soon) conda-forge. I'd say the functionality is now ready for use, with the caveat that it hasn't seen much use by anyone other then me. We do fully pass the scikit-learn test suite for `GridSearchCV` and `RandomizedSearchCV`, so that's at least something. If you do use it, please report any rough edges or bugs you find :).
